clang/include/clang/Sema/Overload.h

```diff
@@ -1005,14 +1005,14 @@ class Sema;
 
     // An overload is a perfect match if the conversion
     // sequences for each argument are perfect.
-    bool isPerfectMatch(const ASTContext &Ctx) const {
+    bool isPerfectMatch(const ASTContext &Ctx, bool ForConversion) const {
```

Comment:
Vastly prefer this take the `Kind`, then line 1015 describe why conversions are ok when CSK_InitByUserDefinedConversion.